features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt

```diff
@@ -382,13 +398,15 @@ internal open class RumViewScope(
 
         if (stopped) return
 
-        interactionToNextViewMetricResolver.onActionSent(
-            InternalInteractionContext(
-                viewId,
-                event.type,
-                event.eventTime.nanoTime
+        if (!event.waitForStop) {
+            interactionToNextViewMetricResolver.onActionSent(
```

Comment:
What about the `ACTION_DROPPED_WARNING` that may occur next in this function?

> RUM Action (%s on %s) was dropped because another action is still active for the same view.

According to RUM rules, if a new action occurs within 100ms after the previous one, the new action is ignored.

However, the way we update the ITNV here seems to disregard this rule. The metric is being updated for every new action, even if the action is skipped due to debouncing. This is problematic, as ITNV could eventually point to a RUM action that does not exist.

To illustrate, if actions A2 and A3 follow A1 immediately, both A2 and A3 will be skipped (not sent). With the current implementation, the next view’s ITNV will be calculated from A3 (`ITNV2(B)`), but A3 will not exist in the RUM. What we expect is `ITNV1(B)`, which should span from the existing action (A1):
```
[===== view A ========][======== view B =========]

  [-A1-]✅             |
    [-A2-]❌           |
      [-A3-]❌         |

-------------------------------------------------->

   ^---- ITNV1(B) -----^
       ^-- ITNV2(B) ---^ 
```

Reply:
You are right but in the same time it might be a problem if we are following these rules. Normally should be the last action that follows to the next view and we should not care about our internal rules to send RUM actions. What do you think ? I was thinking to not add the RUM action rules in this equation as these are only to make sure we dispatch action events reliably.